csharp/src/Google.Protobuf/Collections/RepeatedField.cs

```diff
@@ -101,10 +101,11 @@ public void AddEntriesFrom(CodedInputStream input, FieldCodec<T> codec)
                 int length = input.ReadLength();
                 if (length > 0)
                 {
+                    EnsureSize(count + length);
```

Comment:
Keeping this in mind, **IF** I will have good benchmark results, on real life applications, I will try to create a complete proposal backed by proper analysis and test-cases. I've noticed there are some potential areas to improve, however I am not sure if those improvements would be significant enough.